### PR TITLE
3 get /api/categories

### PR DESCRIPTION
**Pull request by MichaelJRossDev**
Added better testing. I'll add some code to tidy up the running of the server separately on the main branch as it's not related to a specific feature.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const app = require('../app')
+const data = require('../db/data/test-data/index')
+const seed = require('../db/seeds/seed')
+
+const categoriesData = require('../db/data/test-data/categories')
+
+beforeEach(() => {
+    return seed(data);
+})
+
```
Comment (corkesw) [inline __tests__/app.test.js:11]: It might be worth adding an afterAll hook to end your db connection

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+describe('GET Categories', () => {
+    test('should return object with one entry: {"categories" : <all categories data>}', () => {
+        return request(app).get('/api/categories')
+        .expect(200)
+        .then((res) => {
+            const categories = res.body.categories;
+
+            console.log(categories);
```
Comment (corkesw) [inline __tests__/app.test.js:19]: You should remove any console logs

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+            expect(Array.isArray(categories)).toBe(true);
```
Comment (corkesw) [inline __tests__/app.test.js:21]: You can remove this assertion as we implicitly check that with have an array with the forEach later on

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            expect(categories.length).toBe(4);
+            
+            categories.forEach(category => {
+                expect(typeof category.slug).toBe('string');
+                expect(typeof category.description).toBe('string');
+                expect(Object.keys(category).length).toBe(2);
```
Comment (corkesw) [inline __tests__/app.test.js:27]: I wouldn't make this assertion as it makes our test brittle. We're not really bothered about there being additional keys here.

File: controllers/categories.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const { selectCategories } = require('../models/categories');
+
+exports.sendCategories = (req, res) => {
+    selectCategories().then((categories) => res.status(200).send({ categories }));
+};
```
Comment (corkesw) [inline controllers/categories.js:5]: You should have a catch block here in case anything unexpected happens!

**Review by corkesw — COMMENTED** [review]
Great work so far. You might want to add a couple of things here. Firstly, I've suggested adding a catch block to your controller - any caught error will need somewhere to go so I would suggest invoking next and including a default 500 error handler in `app.js`. You will also need to test for and handle an invalid path eg `/api/banana`. This is a single stand alone test that you will need to do once.

File: __tests__/app.test.js (hunk 2 above)
Comment (MichaelJRossDev) [reply __tests__/app.test.js:19]: oops

**Pull request merged**